client/engine/chainservice/eth_chain_helpers.go

```diff
@@ -12,66 +11,69 @@ import (
 
 // getChainHolding reads on-chain holdings for a channel and an asset address given a transaction and an event generated by the transaction.
 func getChainHolding(na *NitroAdjudicator.NitroAdjudicator, tx *types.Transaction, event *NitroAdjudicator.NitroAdjudicatorAllocationUpdated) (common.Address, *big.Int, error) {
-	assetAddress, err := assetAddressForIndex(na, tx, event.AssetIndex)
-	if err != nil {
-		return assetAddress, &big.Int{}, err
-	}
+	// TODO: assetAddressForIndex seems to fail against FEVM.
+	// Since we're just using the native asset we can just use the zero address for now.
+	//	assetAddress, err := assetAddressForIndex(na, tx, event.AssetIndex)
+	// if err != nil {
+	// 	return assetAddress, &big.Int{}, err
+	// }
+	assetAddress := common.Address{}
 	amount, err := na.Holdings(&bind.CallOpts{BlockNumber: new(big.Int).SetUint64(event.Raw.BlockNumber)}, assetAddress, event.ChannelId)
 	if err != nil {
 		return assetAddress, amount, err
 	}
 	return assetAddress, amount, nil
 }
 
-// assetAddressForIndex uses the input parameters of a transaction to map an asset index to an asset address
-func assetAddressForIndex(na *NitroAdjudicator.NitroAdjudicator, tx *types.Transaction, index *big.Int) (common.Address, error) {
-	abi, err := NitroAdjudicator.NitroAdjudicatorMetaData.GetAbi()
-	if err != nil {
-		return common.Address{}, err
-	}
-	params, err := decodeTxParams(abi, tx.Data())
```

Comment:
It seems that the `tx.Data()` call fails when targeting the FEVM. For now I've just commented out the code.